### DISCUSSION TITLE
Can I use this package on the whole image?

### DISCUSSION THREAD

**Discussion by pierluigi-rinaldi** [PetroFit/petrofit#76] [Questions]
Hi All, 

I was looking for a package to create a model for my image and I found this one.
I have a very ultra-deep image (a simulated one) where I want to model bright sources on the field of view to subtract them and extract the faintest ones. I wonder if I can use this package to create a model image using my whole image (let's say 4000x4000) or it only works on a very little piece of it.

Thank you in advance!

**Comment by robelgeda** [comment]
Hi @x12red,  you can absolutely model the whole image but since the run time is a function of the image size, it may take longer to finish fitting. 

The long answer is that it depends on what goes into the model and if you can share, I can provide some advice. If you have a PSF that convolves the image, the convolution is what takes the longest time. It also depends on whether your sources are overlapping. The reason I ask is because there are several techniques to speed up the process. Lastly, I am in the middle of upgrading the code and if you can describe what the ideal workflow looks like, we can try to implement it. 

Thanks for opening a ticket!

**Comment by robelgeda** [comment]
I did a test with an image with size 4000 pixels and a PSF of size 50 pixels, I made a single Sersic profile as the simulated data. I then fitted the image with reasonable parameter guesses.  I ran the fit on the default settings, except I set the accuracy to a much higher accuracy of `acc=1e-17`. This test ran in less than 3 minutes. Your PSF is probably much larger than 50 pixels but I hope this provides an idea of what the performance will look like.

**Comment by pierluigi-rinaldi** [comment]
Dear Robel, 

Thanks for your quick reply and tests.
I can share with you my image if you give me an email. Please, let me know.

Basically, my idea would be the following one:
I have this image with ~12000 sources. Most of them are very faint sources (up to mag ~ 30/31). I would run a source extraction code (let's say SExtractor) in a very cold mode to find the brightest sources only. Once I find them, I would model my image to remove those bright sources and run SExtractor again. And so on 'till I reach good completeness.  I also have a simulated PSF, if it can help.

**Comment by robelgeda** [comment]
I can take a look at the data and give feedback. My email is on my [GitHub page](https://github.com/robelgeda) and if you upload the data to the cloud, I can download it.

 I would say you should use [PhotUtils](https://photutils.readthedocs.io/en/stable/segmentation.html) for your segmentation and deblending. If you are experienced with PhotUtils, PetroFit provides [shortcut functions](https://petrofit.readthedocs.io/en/latest/photometry_and_petrosian.html#Catalogs) to do this. I would recommend playing around with the PhotUtils functions if you haven't had the chance to use it yet.

**Comment by robelgeda** [comment]
Converting this issue into a discussion. #76

**Comment by robelgeda** [comment]
@x12red We can continue to chat here.
